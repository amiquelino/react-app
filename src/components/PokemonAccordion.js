import React from 'react'
import {Panel, Grid, Row, Col, Badge} from 'react-bootstrap'
import PropTypes from 'prop-types'

const PokemonAccordion = ( props ) => {
  return (
    <div>
      {props.dataset.map((pokemon) => {
        return (
          <Panel  collapsible expanded={props.selected.name && props.selected.name === pokemon.name}
            header={pokemon.name}
            eventKey={pokemon.url}
            key={pokemon.name}
            onSelect={props.onSelect}>
            <Grid>
              <Row className="show-grid">
                <Col xs={3} md={2}><b>{'Nome: '}</b></Col>
                <Col xs={2} md={1}><b>{'CP: '}</b></Col>
                <Col xs={2} md={1}><b>{'Altura: '}</b></Col>
                <Col xs={2} md={1}><b>{'Peso: '}</b></Col>
                <Col xs={2} md={1}><b>{'Ordem: '}</b></Col>
              </Row>
              <Row className="show-grid">
                <Col xs={3} md={2}>{props.selected.name}</Col>
                <Col xs={2} md={1}><Badge>{props.selected.base_experience}</Badge></Col>
                <Col xs={2} md={1}><Badge>{props.selected.height}</Badge></Col>
                <Col xs={2} md={1}><Badge>{props.selected.weight}</Badge></Col>
                <Col xs={2} md={1}><Badge>{props.selected.order}</Badge></Col>
              </Row>
            </Grid>
         </Panel>
         
        )
      })}
    </div>
  )
}

PokemonAccordion.propTypes = {
  dataset: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired,
}




export default PokemonAccordion
