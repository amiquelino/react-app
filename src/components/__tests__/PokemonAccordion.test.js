import React from 'react'
import renderer from 'react-test-renderer'
import PokemonAccordion from '../PokemonAccordion'

describe('PokemonAccordion', ( ) => {
  const pokemons = [
    {
      "url" : "http://pokeapi.co/api/v2/pokemon/1/",
      "name":"bulbasaur"
    },
    {
      "url" : "http://pokeapi.co/api/v2/pokemon/2/",
      "name" : "ivysaur"
    }
  ]

  it('render correctly', ( ) => {
    const component = renderer.create(
      <PokemonAccordion
        dataset={pokemons}
      />
    )
    expect( component ).toMatchSnapshot( )
  })
})
