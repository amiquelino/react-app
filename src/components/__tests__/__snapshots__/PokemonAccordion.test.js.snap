exports[`PokemonAccordion render correctly 1`] = `
<div>
  <div
    className="panel panel-default"
    id={null}>
    <div
      className="panel-heading">
      <h4
        className="panel-title"
        role="presentation">
        <a
          aria-controls={undefined}
          aria-expanded={true}
          aria-selected={true}
          className={null}
          href={undefined}
          onClick={[Function]}
          role={undefined}>
          bulbasaur
        </a>
      </h4>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={false}
      className="panel-collapse collapse in"
      id={undefined}
      role={undefined}>
      <div
        className="panel-body">
        <div
          className="container">
          <div
            className="show-grid row">
            <div
              className="col-md-2 col-xs-3">
              <b>
                Nome: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                CP: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Altura: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Peso: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Ordem: 
              </b>
            </div>
          </div>
          <div
            className="show-grid row">
            <div
              className="col-md-2 col-xs-3">
              bulbasaur
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                64
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                7
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                69
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={null}>
    <div
      className="panel-heading">
      <h4
        className="panel-title"
        role="presentation">
        <a
          aria-controls={undefined}
          aria-expanded={false}
          aria-selected={false}
          className="collapsed"
          href={undefined}
          onClick={[Function]}
          role={undefined}>
          ivysaur
        </a>
      </h4>
    </div>
    <div
      aria-expanded={null}
      aria-hidden={true}
      className="panel-collapse collapse"
      id={undefined}
      role={undefined}>
      <div
        className="panel-body">
        <div
          className="container">
          <div
            className="show-grid row">
            <div
              className="col-md-2 col-xs-3">
              <b>
                Nome: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                CP: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Altura: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Peso: 
              </b>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <b>
                Ordem: 
              </b>
            </div>
          </div>
          <div
            className="show-grid row">
            <div
              className="col-md-2 col-xs-3">
              bulbasaur
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                64
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                7
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                69
              </span>
            </div>
            <div
              className="col-md-1 col-xs-2">
              <span
                className="badge">
                1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
